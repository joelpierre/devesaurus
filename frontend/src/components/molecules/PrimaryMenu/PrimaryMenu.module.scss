@import "~sass/core";

.primary-menu {
  $root: &;

  display: inline-flex;
  width: auto;

  &__list {
    list-style-type: none;
    margin-bottom: 0;
  }

  &__item {
    display: inline-block;
    margin-bottom: 0;
  }
}
