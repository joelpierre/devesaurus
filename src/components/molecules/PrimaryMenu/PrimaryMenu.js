import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import PropTypes from 'prop-types';

import styles from './PrimaryMenu.module.scss';

import MenuItem from '../../atoms/MenuItem/MenuItem';

export const PurePrimaryMenu = ({
  classes,
  allWordpressWpApiMenusMenusItems,
}) => (
  <nav
    data-test="component-primary-menu"
    className={`${styles['primary-menu']} ${classes}`}
  >
    <ul className={styles['primary-menu__list']}>
      {allWordpressWpApiMenusMenusItems.edges[0].node.items.map(item => (
        <MenuItem
          classes={styles['primary-menu__item']}
          key={item.object_slug}
          item={item}
        />
      ))}
    </ul>
  </nav>
);

const PrimaryMenu = ({ classes }) => (
  <StaticQuery
    data-test="component-primary-menu-query"
    query={menuQuery}
    render={props => <PurePrimaryMenu {...props} classes={classes} />}
  />
);

PrimaryMenu.defaultProps = {
  classes: null,
};

PrimaryMenu.propTypes = {
  classes: PropTypes.string,
};

export default PrimaryMenu;

const menuQuery = graphql`
  {
    allWordpressWpApiMenusMenusItems(filter: { name: { eq: "Primary Menu" } }) {
      edges {
        node {
          name
          items {
            order
            title
            object_slug
          }
        }
      }
    }
  }
`;
