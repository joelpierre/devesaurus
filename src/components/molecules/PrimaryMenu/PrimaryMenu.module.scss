@import "~sass/core";

.primary-menu {
  $root: &;

  display: flex;
  align-items: center;
  width: auto;

  &__list {
    display: flex;
    list-style-type: none;
    margin-bottom: 0;
  }

  &__item {
    display: inline-block;
    margin-bottom: 0;
  }
}
