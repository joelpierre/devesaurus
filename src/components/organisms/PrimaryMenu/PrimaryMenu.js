import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';

import styles from './PrimaryMenu.module.scss';
import * as coreActions from '../../../store/actions/core.actions';
import MenuItem from '../../atoms/MenuItem/MenuItem';

export const PurePrimaryMenu = ({
  classes,
  isMenuOpen,
  allWordpressWpApiMenusMenusItems,
  setMenuState,
}) => {
  const toggleMenuState = () => {
    setMenuState(!isMenuOpen);
  };

  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions
    <nav
      data-test="component-primary-menu"
      className={classNames(
        styles['primary-menu'],
        { [styles['primary-menu--is-active']]: isMenuOpen },
        classes
      )}
      onClick={toggleMenuState}
    >
      <ul className={styles['primary-menu__list']}>
        {allWordpressWpApiMenusMenusItems.edges[0].node.items.map(item => (
          <MenuItem
            classes={styles['primary-menu__item']}
            key={item.object_slug}
            item={item}
          />
        ))}
      </ul>
    </nav>
  );
};

export const PrimaryMenu = ({ classes, isMenuOpen, setMenuState }) => (
  <StaticQuery
    data-test="component-primary-menu-query"
    query={menuQuery}
    render={props => (
      <PurePrimaryMenu
        {...props}
        classes={classes}
        isMenuOpen={isMenuOpen}
        setMenuState={setMenuState}
      />
    )}
  />
);

PrimaryMenu.defaultProps = {
  classes: undefined,
};

PrimaryMenu.propTypes = {
  classes: PropTypes.string,
  isMenuOpen: PropTypes.bool.isRequired,
  setMenuState: PropTypes.func.isRequired,
};

/* istanbul ignore next */
const mapStateToProps = ({ core: { isMenuOpen } }) => ({
  isMenuOpen,
});

/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  setMenuState: data => dispatch(coreActions.setMenuState(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PrimaryMenu);

const menuQuery = graphql`
  {
    allWordpressWpApiMenusMenusItems(filter: { name: { eq: "Primary Menu" } }) {
      edges {
        node {
          name
          items {
            order
            title
            object_slug
          }
        }
      }
    }
  }
`;
