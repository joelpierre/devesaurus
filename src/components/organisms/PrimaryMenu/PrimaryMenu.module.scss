@import "~sass/core";

.primary-menu {
  $root: &;

  display: flex;
  align-items: flex-start;
  width: 100%;
  position: fixed;
  height: 100vh;
  top: 0;
  left: -100%;
  z-index: 10;
  transition: .8s all;
  overflow: hidden;

  &--is-active {
    left: 0;
  }

  &__overlay {
    appearance: none;
    width: 100%;
    height: 100%;
    border: none;
    box-shadow: none;
    display: block;
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    cursor: pointer;
    z-index: -50;
    background-color: rgba($color-tint-psi, .6);
    opacity: 0;
    visibility: hidden;

    &--is-active {
      opacity: 1;
      z-index: 50;
      visibility: visible;
    }

    @include hover-focus {
      background-color: rgba($color-tint-psi, .4);
    }
  }

  &__brand {
    margin: auto;
    width: auto;
  }

  &__inner {
    position: relative;
    width: 450px;
    height: 100%;
    z-index: 100;
    background-color: $color-tint-beta;
    padding: 2rem;
  }

  &__list {
    display: flex;
    flex-wrap: wrap;
    list-style-type: none;
    align-items: flex-start;
    align-content: flex-start;
    margin-top: 2rem;
    margin-bottom: 0;
    width: 100%;
    height: 100%;
    z-index: 200;
  }

  &__divider {
    width: 100%;
    height: 2px;
    background: $color-brand;
    margin-top: 1rem;
    margin-bottom: 1rem;
    display: block;
  }

  &__item {
    width: 100%;
    display: flex;
    flex-wrap: nowrap;
    align-items: center;
    margin-bottom: 0;
    font-size: 2rem;
    font-weight: bold;
    text-transform: uppercase;

    & > a {
      display: flex;
      flex-wrap: nowrap;
      align-items: center;
      padding: 2rem 0;
      transition: $transition-all-base;
      margin-right: 2rem;
      text-decoration: none;
    }

    @include hover-focus {
      & > a {
        margin-right: 0;
      }
    }
  }

  &__icon {
    margin-left: auto;
  }
}
