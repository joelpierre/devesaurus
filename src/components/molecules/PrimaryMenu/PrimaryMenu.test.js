import React from 'react';
import { shallow } from 'enzyme';
import { StaticQuery } from 'gatsby';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import PrimaryMenu, { PurePrimaryMenu } from './PrimaryMenu';

const defaultProps = {};

const defaultPureProps = {
  allWordpressWpApiMenusMenusItems: {
    edges: [
      {
        node: {
          items: [
            {
              title: 'title',
              object_slug: 'slug',
            },
          ],
        },
      },
    ],
  },
};

/**
 * Factory function to create a ShallowWrapper for the PrimaryMenu component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<PrimaryMenu {...setupProps} />);
};

const pureSetup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<PurePrimaryMenu {...setupProps} />);
};

describe('<PrimaryMenu/>', () => {
  let wrapper;
  let pureWrapper;

  beforeEach(() => {
    wrapper = setup();
    pureWrapper = pureSetup();
  });

  it('renders the PrimaryMenu Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-primary-menu-query');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(PrimaryMenu, defaultProps);
  });

  it('should render with correct props', () => {
    checkProps(PurePrimaryMenu, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should match snapshot', () => {
    matchSnapshot(pureWrapper);
  });
});
