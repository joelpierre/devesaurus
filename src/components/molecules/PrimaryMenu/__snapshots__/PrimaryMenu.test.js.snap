// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryMenu/> should match snapshot 1`] = `
<mockConstructor
  data-test="component-primary-menu-query"
  render={[Function]}
/>
`;

exports[`<PrimaryMenu/> should match snapshot 2`] = `
<nav
  className="primary-menu undefined"
  data-test="component-primary-menu"
>
  <ul
    className="primary-menu__list"
  >
    <MenuItem
      classes="primary-menu__item"
      item={
        Object {
          "object_slug": "slug",
          "title": "title",
        }
      }
      key="slug"
    />
  </ul>
</nav>
`;
